src/transform/transform.ts

```diff
@@ -518,65 +517,51 @@ function processOperationResponses(session: Session<CodeModel>) {
   }
   for (const group of values(session.model.operationGroups)) {
     for (const op of values(group.operations)) {
-      // annotate all exception types as errors; this is so we know to generate an Error() method
-      const exTypeNames = new Array<string>();
-      for (const ex of values(op.exceptions)) {
-        const marshallingFormat = getMarshallingFormat(ex.protocol);
-        if (marshallingFormat === 'na') {
-          // this is for the case where the 'default' response section
-          // doesn't specify a model (legal, mostly in the test server)
-          ex.language.go!.genericError = true;
-          continue;
-        }
-        const schemaError = (<SchemaResponse>ex).schema;
-        if (isObjectSchema(schemaError)) {
-          for (const prop of values(schemaError.properties)) {
-            // adding the Inner prefix on error types, since errors in Go have an Error() method
-            // in order to implement the error interface. This causes errors to not be able to have
-            // an Error field as well, since it would cause confusion
-            if (prop.language.go!.name === 'Error') {
-              prop.language.go!.name = 'Inner' + prop.language.go!.name;
+      // TODO: make this work once relevant M4 bugs have been fixed
+      // annotate all exception types as errors so we can skip generating them
```

Comment:
Do you want to link the issues here?

Reply:
I haven't dug into this fully yet and opened them as Tim was OOF at the time.  I'm tracking this overall work in https://github.com/Azure/autorest.go/issues/730